test/e2e/operator_test.go

```diff
@@ -1284,3 +1420,52 @@ func stsModeRequested() bool {
 	}
 	return false
 }
+
+// extractArg extracts the value of a specific argument from a list of container arguments.
+// Returns an error if the argument is present but malformed, or if not found.
+func extractArg(args []string, argKey string) (string, error) {
+	for _, arg := range args {
+		if strings.HasPrefix(arg, argKey+"=") {
+			parts := strings.SplitN(arg, "=", 2)
+			if len(parts) != 2 {
+				return "", fmt.Errorf("invalid format for argument %s: %s", argKey, arg)
+			}
+			return parts[1], nil
+		}
+	}
+	return "", fmt.Errorf("argument %s not found", argKey)
+}
+
+// assertContainerArgFromDeployment asserts that a container in a deployment has a
+// specific argument with an expected value.
+func assertContainerArgFromDeployment(t *testing.T, dep *appsv1.Deployment, containerName, argKey, expectedArgValue string) {
+	t.Helper()
+	t.Logf("Asserting container %q has argument %q with value %q", containerName, argKey, expectedArgValue)
+	for _, c := range dep.Spec.Template.Spec.Containers {
+		if c.Name == containerName {
+			actualArgValue, err := extractArg(c.Args, argKey)
+			if err != nil {
+				t.Fatalf("failed to extract argument %q for container %q: %v", argKey, containerName, err)
+			}
+
+			if actualArgValue != expectedArgValue {
+				t.Fatalf("container %q, argument %q: expected value %q, but got %q", containerName, argKey, expectedArgValue, actualArgValue)
+			}
+			return
+		}
+	}
+	t.Fatalf("container %q not found in deployment", containerName)
+}
+
+func isManagedServiceCluster(ctx context.Context, adminClient kubernetes.Interface) (bool, error) {
+	_, err := adminClient.CoreV1().Namespaces().Get(ctx, "openshift-backplane", v1.GetOptions{})
+	if err == nil {
+		return true, nil
+	}
+
+	if !errors.IsNotFound(err) {
+		return false, err
+	}
+
+	return false, nil
+}
```

Comment:
This check is inspired by https://github.com/openshift/origin/pull/29216/files#diff-35a89a7a7362642eebb559fb8564e857b00d6f7dd6322c3adabaf1adbd609d35R2267-R2278 implementation.
xRef: [Slack discussion](https://redhat-external.slack.com/archives/C01C8502FMM/p1731001251801799?thread_ts=1730974270.586539&cid=C01C8502FMM)

Reply:
Sure, added the PR link as code comment